### PR TITLE
WEBUMENIA-1475 laravel-mix for JavaScript

### PR DESCRIPTION
**Pull request by eronisko**
# Description

I've moved assets from the 'master' layout into a laravel-mix Webpack pipeline. It'll take a while to move everthing else over but I'm hoping this'll already give use some performance benefits and form a base for future javascript code.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: resources/js/components/searchbar.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
+require('typeahead.js/dist/typeahead.bundle')
```
Comment (eronisko) [inline resources/js/components/searchbar.js:1]: Note there's nothing new here in this file. I've just stripped off the `<script>` tags and added `require`.

File: resources/views/intro.blade.php [hunk 2]
```diff
@@ -110,7 +110,7 @@
     });
     $carousel.children('.flickity-page-dots').css('left',  parseInt($('.flickity-slider').css('transform').split(',')[4]) );
 
-    $carousel.on( 'staticClick', function( event, pointer, cellElement, cellIndex ) {
+    $carousel.on('staticClick.flickity', function( event, pointer, cellElement, cellIndex ) {
```
Comment (eronisko) [inline resources/views/intro.blade.php:113]: Required for upgrade to v2 (which is compatible with Webpack)

Comment (martin-havala) [reply resources/views/intro.blade.php:113]: flickity is going to be removed by slick... under redesign bunch

Comment (eronisko) [reply resources/views/intro.blade.php:113]: I'm bearing that in mind but I've had to make this work with current develop. I'll be happy to help you with that once you merge in the design branch.

Comment (eronisko) [reply resources/views/intro.blade.php:113]: Or (of course) I'll be happy to adjust it this if design is merged in first.

**Review by martin-havala — APPROVED** [review]

**Comment by igor-kamil** [comment]
this is great! 📈

**Pull request merged**